src/components/AddVotes.jsx

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useState, useEffect } from "react";
+import { patchVotes } from "../api";
+
+export const AddVotes = (props) => {
+  const [votesChange, setVotesChange] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const incVotes = useCallback(async () => {
+    setIsLoading(true);
+    await patchVotes(props.article_id, votesChange + props.currentVotes);
+
+    setVotesChange(votesChange + 1);
+    setIsLoading(false);
+  }, [votesChange, props.article_id, props.currentVotes]);
```

Comment:
Nice to see you trying out some other React hooks here. `useCallback` is performance optimisation though and it's really not needed here. Works all the same but the code would be simpler if you just used a regular function. Feel free to leave it in if you're just figuring out how it works 👍